src/lib.rs

```diff
@@ -0,0 +1,389 @@
+//! `fm` is a simple non-backtracking fuzzy text matcher useful for matching multi-line patterns
+//! and text. At its most basic the wildcard operator (`...` by default) can be used in the
+//! following ways:
+//!
+//!   * If a line consists solely of `...` it means "match zero or more lines of text".
+//!   * If a line starts with `...` it means "match the remainder of the line only".
+//!   * If a line ends with `...`, it means "match the start of the line only".
+//!   * If a line starts and ends with `...` it means "match the inner part of the line".
+//!
+//! If the wildcard operator appears in any other locations, it is matched literally. Wildcard
+//! matching does not backtrack, so if a line consists solely of `...` then the next matching
+//! line "anchors" the remainder of the search.
+//!
+//! The following examples show `fm` in action using its defaults (i.e. `...` as the wildcard
+//! operator, and leading/trailing whitespace ignored):
+//!
+//! ```rust
+//! use fm::FMatcher;
+//!
+//! assert!(FMatcher::new("a").unwrap().matches("a").is_ok());
+//! assert!(FMatcher::new(" a ").unwrap().matches("a").is_ok());
+//! assert!(FMatcher::new("a").unwrap().matches("b").is_err());
+//! assert!(FMatcher::new("a\n...\nb").unwrap().matches("a\na\nb").is_ok());
+//! assert!(FMatcher::new("a\n...\nb").unwrap().matches("a\na\nb\nb").is_err());
+//! ```
+//!
+//! If you want to use non-default options, you will first need to use `FMBuilder` before having
+//! access to an `FMatcher`.
```

Comment:
Did you want to mention the placeholder support that you use in the tests?

Reply:
Placeholder?

Reply:
I think you are calling it "Names"?

Reply:
You've completely lost me :)

Reply:
Your name matcher. It's one of the best features of the library. Why not flaunt it here in the module overview?